node/resolvers/checkout/index.ts

```diff
@@ -321,6 +322,28 @@ export const queries: Record<string, Resolver> = {
       s => path(['pickupStoreInfo', 'address', 'addressId'], s) === pickupId
     )
   },
+
+  itemsWithSimulation: async (_, { items }: any, ctx: Context) => {
+    const {
+      clients: { checkout },
+      vtex: { segment }
+    } = ctx
+
+    const simulationPayloads: SimulationPayload[] = getSimulationPayloads(items, segment?.priceTables, segment?.regionId)
+    const simulationPromises = simulationPayloads.map(payload => checkout.simulation(payload))
+    const simulations = await Promise.all(simulationPromises)
+
+    const simulationItems = simulations.map(simulation => simulation.items.map((item: any) => ({ ...item, paymentData: simulation.paymentData }))).reduce((acc, val) => acc.concat(val), [])
+    const groupedBySkuId = groupBy(prop('id'), simulationItems)
+
+    const orderItemsBySellerById: OrderFormItemBySellerById = mergeAll(Object.entries(groupedBySkuId).map(([id, items]) => {
+      const groupedBySeller = indexBy((prop('seller')), items)
+
+      return { [id]: groupedBySeller }
+    }))
+
+    return items.map((item: Item) => fillSearchItemWithSimulation(item, orderItemsBySellerById[item.itemId]))
```

Comment:
Can we change this code to avoid iterating through the items so many times?

Also avoiding using ramda

Reply:
This code was copied and pasted from another place. It took me a while to understand and rewrite it. But it's done. 🙏 

I removed all the unnecessary loops and improved the readability.